test/reduce.cpp

```diff
@@ -312,6 +318,969 @@ CPP_REDUCE_TESTS(anyTrue, any_true, float, unsigned char);
+            rbk_unique_data<Tk, Tv>("multiblock", 1024, generateConsq<Tk>(0),
+                                    generateConst<Tv>(1)),
+            rbk_unique_data<Tk, Tv>("multiblock1", 1025, generateConsq<Tk>(0),
+                                    generateConst<Tv>(1)),
+            rbk_unique_data<Tk, Tv>("large", 1024 * 1025, generateConsq<Tk>(0),
+                                    generateConst<Tv>(1))};
+}
+
+template <typename Tk, typename Tv>
+vector<reduce_by_key_params<Tk, Tv> > genSingleKeyTests() {
+    return {rbk_single_data<Tk, Tv>("subwarp", 31, generateConst<Tk>(0),
+                                    generateConst<Tv>(1)),
+            rbk_single_data<Tk, Tv>("warp", 32, generateConst<Tk>(0),
+                                    generateConst<Tv>(1)),
+            rbk_single_data<Tk, Tv>("warp1", 33, generateConst<Tk>(0),
+                                    generateConst<Tv>(1)),
+            rbk_single_data<Tk, Tv>("subblock", 127, generateConst<Tk>(0),
+                                    generateConst<Tv>(1)),
+            rbk_single_data<Tk, Tv>("block", 128, generateConst<Tk>(0),
+                                    generateConst<Tv>(1)),
+            rbk_single_data<Tk, Tv>("block1", 129, generateConst<Tk>(0),
+                                    generateConst<Tv>(1)),
+            rbk_single_data<Tk, Tv>("multiblock", 1024, generateConst<Tk>(0),
+                                    generateConst<Tv>(1)),
+            rbk_single_data<Tk, Tv>("multiblock1", 1025, generateConst<Tk>(0),
+                                    generateConst<Tv>(1)),
+            rbk_single_data<Tk, Tv>("large", 128 * 1025, generateConst<Tk>(0),
+                                    generateConst<Tv>(1))};
+}
+
+template <typename TestClass>
+string testNameGenerator(
+    const ::testing::TestParamInfo<typename TestClass::ParamType> info) {
+    return info.param.testname_;
+}
+
+INSTANTIATE_TEST_CASE_P(UniqueKeyTests, ReduceByKey_if,
+                        ::testing::ValuesIn(genUniqueKeyTests<int, float>()),
+                        testNameGenerator<ReduceByKey_if>);
+
+INSTANTIATE_TEST_CASE_P(SingleKeyTests, ReduceByKey_if,
+                        ::testing::ValuesIn(genSingleKeyTests<int, float>()),
+                        testNameGenerator<ReduceByKey_if>);
+TEST_P(ReduceByKey_if, UniqueKeyTests) {
+    reduce_by_key_params<int, float> params = GetParam();
+
+    array keys(params.iKeys_.size(), params.iKeys_.data());
+    array vals(params.iVals_.size(), params.iVals_.data());
+
+    array keyResGold(params.oKeys_.size(), params.oKeys_.data());
+    array valsReducedGold(params.oVals_.size(), params.oVals_.data());
+
+    array keyRes, valsReduced;
+    sumByKey(keyRes, valsReduced, keys, vals, 0, 0);
+
+    ASSERT_TRUE(allTrue<bool>(keyResGold == keyRes));
+    ASSERT_ARRAYS_NEAR(valsReduced, valsReducedGold, 1e-5);
+}
+
+TEST_P(ReduceByKey_if, SingleKeyTests) {
+    reduce_by_key_params<int, float> params = GetParam();
```

Comment:
Can't any one of the following combinations be merged some how  ?  they seem to have similar code except for the code stubs inside the `TEST_P` definitions.
- one combination `TEST_P(ReduceByKey_if, UniqueKeyTests)` & `
TEST_P(ReduceByKey_if, SingleKeyTests)`
- another combination `
TEST_P(ReduceByKey_if, SingleKeyTests)` & `TEST_P(ReduceByKey_uf, SingleKeyTests)`
I guess this more like can we do that easily in google test ?